webapp/src/App.tsx

```diff
@@ -50,73 +50,95 @@ export const useClasses = makeStyles({
 enum AppState {
     ProbeForBackend,
     SettingUserInfo,
+    ErrorLoadingAuthInfo,
+    LoadingAuthInfo,
+    ErrorLoadingChats,
     ErrorLoadingUserInfo,
     LoadingChats,
     Chat,
     SigningOut,
 }
 
-const App: FC = () => {
+const App = () => {
     const classes = useClasses();
 
-    const [appState, setAppState] = React.useState(AppState.ProbeForBackend);
+    const [appState, setAppState] = React.useState(AppState.LoadingAuthInfo);
     const dispatch = useAppDispatch();
 
     const { instance, inProgress } = useMsal();
-    const { activeUserInfo, features, isMaintenance } = useAppSelector((state: RootState) => state.app);
+    const { authConfig, features, isMaintenance } = useAppSelector((state: RootState) => state.app);
     const isAuthenticated = useIsAuthenticated();
 
     const chat = useChat();
     const file = useFile();
 
+    useEffect(() => {
+        // if the auth info is being loaded, change the state if:
+        //      1. the `authConfig` is falsy, meaning an error occurred during load, or
+        //      2. the `authConfig` has keys, meaning it was loaded successfully.
+        if (appState === AppState.LoadingAuthInfo && (!authConfig || Object.keys(authConfig).length)) {
```

Comment:
This setup bypasses the ProbeForBackend state altogether unless we hit a specific error case, but we need to hit the ProbeForBackend state at least once so the maintenceCheck can be hit. 

It might make more sense to Load the AuthInfo in the BackendProbe component in this series of events:
1. LoadAuthInfo
2. then Maintenace Check

If both 1 and 2 pass, then you can bypass the /healthz check there (which is how it's setup now where MaintainenceCheck is queried and the /healthz probe is bypassed if /maintencestatus is hit)

Reply:
thanks for calling out the maintenance check. I've updated `App` to always begin on the `ProbeForBackend` state, can you take another look?

I ran into issues loading the auth info in `BackendProbe`, since the auth info is used to render the `MsalProvider`, which needs to sit above the `App` in the React tree for the hooks to work. I experimented with having the `BackendProbe` be the top-level- component in `index.tsx`, but it led to too much polling and some other weirdness.